.github/workflows/ci_tests.yml

```
@@ -12,63 +12,90 @@ on:
+          - platform: ubuntu-latest
+            compiler:
+              cpp: g++
+              c: gcc
+            cmake_args:
+              description: "A-San"
+              cmake_args: "-DCMAKE_CXX_FLAGS=-fsanitize=address -fsanitize=undefined"
+          - platform: ubuntu-latest
+            compiler:
+              cpp: g++
+              c: gcc
+            cmake_args:
+              description: "T-San"
+              cmake_args: "-DCMAKE_CXX_FLAGS=-fsanitize=thread"
 
+    name: "Bulid & Test: ${{ matrix.compiler.c }} ${{ matrix.cmake_args.description }}"
+    runs-on: ${{ matrix.platform }}
     steps:
       - uses: actions/checkout@v4
-      # GitHub runners have updated the Ubuntu Linux Kernel to use strong ASLR,
-      # but LLVM is not configured for this change, and thus the address
-      # sanitizer breaks.
-      #
-      # The next image is supposed to fix this, so if the Ubuntu image has been
-      # updated, this work around is no longer required.
-      - name: get runner image version
-        id: runner-image-version
+      - name: Install Ninja
+        uses: lukka/get-cmake@latest
+        with:
+          cmakeVersion: "~3.25.0"
+          ninjaVersion: "^1.11.1"
+      - name: Print installed softwares
         run: |
-          echo "image-version=$(echo $ImageVersion)" >> "$GITHUB_OUTPUT"
-        working-directory: .
-      - name: modify number of bits to use for ASLR entropy
-        if: ${{ steps.runner-image-version.outputs.ImageVersion }} == '20240310.1.0'
+          clang++ --version
+          g++ --version
+          cmake --version
+          ninja --version
+      - name: Build Release
         run: |
-          sudo sysctl -a | grep vm.mmap.rnd
-          sudo sysctl -w vm.mmap_rnd_bits=28
-        working-directory: .
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-      - name: Build and push
-        uses: docker/build-push-action@v6
-        with:
-          context: .
-          push: false
-          load: true
-          tags: ${{ matrix.cfg.id }}
-          file: .ci/docker/${{ matrix.cfg.platform }}.Dockerfile
-          build-args: |
-            cc=${{ matrix.cfg.cc }}
-            cxx=${{ matrix.cfg.cpp }}
-            cmake_args=${{ matrix.cfg.cmake_args }}
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-      - name: Run tests
+          cmake -B build -S . "${{ matrix.cmake_args.args }}"
+          cmake --build build --config Release --verbose
+          cmake --build build --config Release --target all_verify_interface_header_sets
+          cmake --install build --config Release --prefix /opt/beman.exemplar
+          find /opt/beman.exemplar -type f
+        env:
+          CC: ${{ matrix.compiler.c }}
+          CXX: ${{ matrix.compiler.cpp }}
+          CMAKE_GENERATOR: "Ninja Multi-Config"
+      - name: Test Release
+        run: ctest --test-dir build -C Release
```

Prefer long options for readability.

```suggestion
        run: ctest --test-dir build --build-config Release
```

---

Thanks!